rest-api/test/unit_test/api_test/biobank_order_api_test.py

```diff
@@ -17,9 +19,205 @@ def setUp(self):
+    self.assertEqual(get_cancelled_order['amendedReason'], 'Its all wrong')
+    self.assertEqual(get_cancelled_order['cancelledUsername'], 'fred@pmi-ops.org')
+    self.assertEqual(get_cancelled_order['cancelledSiteId'], 1)
+
+  def test_retore_an_order(self):
+    self.summary_dao.insert(self.participant_summary(self.participant))
+    order_json = load_biobank_order_json(self.participant.participantId,
+                                         filename='biobank_order_2.json')
+    result = self.send_post(self.path, order_json)
+    full_order_json = load_biobank_order_json(self.participant.participantId,
+                                              filename='biobank_order_1.json')
+    _strip_fields(result)
+    _strip_fields(full_order_json)
+    self.assertEquals(full_order_json, result)
+
+    biobank_order_id = result['identifier'][1]['value']
+    path = self.path + '/' + biobank_order_id
+    request_data = {
+      "amendedReason": "Its all wrong",
+      "cancelledInfo": {
+        "author": {
+          "system": "https://www.pmi-ops.org/healthpro-username",
+          "value": "fred@pmi-ops.org"
+        },
+        "site": {
+          "system": "https://www.pmi-ops.org/site-id",
+          "value": "hpo-site-monroeville"
+        }
+      },
+      "status": "cancelled"
+    }
+    self.send_patch(path, request_data=request_data, headers={'If-Match': 'W/"1"'})
+    cancelled_order = self.send_get(path)
+    self.assertEqual(cancelled_order['version'], 2)
+
+    request_data = {
+      "amendedReason": "I didnt mean to cancel",
+      "restoredInfo": {
+        "author": {
+          "system": "https://www.pmi-ops.org/healthpro-username",
+          "value": "fred@pmi-ops.org"
+        },
+        "site": {
+          "system": "https://www.pmi-ops.org/site-id",
+          "value": "hpo-site-monroeville"
+        }
+      },
+      "status": "restored"
+    }
+
+    self.send_patch(path, request_data=request_data, headers={'If-Match': 'W/"2"'})
+    restored_order = self.send_get(path)
+    self.assertEqual(restored_order['status'], 'RESTORED')
+    self.assertEqual(restored_order['restoredUsername'], 'fred@pmi-ops.org')
+    self.assertEqual(restored_order['restoredSiteId'], 1)
+    self.assertEqual(restored_order['version'], 3)
+    self.assertEqual(restored_order['amendedReason'], 'I didnt mean to cancel')
+
+  def test_amending_an_order(self):
+    # pylint: disable=unused-variable
+    self.summary_dao.insert(self.participant_summary(self.participant))
+    order_json = load_biobank_order_json(self.participant.participantId,
+                                         filename='biobank_order_2.json')
+    result = self.send_post(self.path, order_json)
+    load_biobank_order_json(self.participant.participantId,
+                            filename='biobank_order_1.json')
+
+    biobank_order_id = result['identifier'][1]['value']
+    path = self.path + '/' + biobank_order_id
+    request_data = {
```

Comment:
Make sure to modify either the identifiers or samples in each amendment, and verify the result... that's the only reason to do an amendment in the first place.

Reply:
I don't see any amendments of identifiers in the tests?

Reply:
It changes the site and removes one sample, would you like more ?

Reply:
I was talking about biobank order identifiers, specifically. Are the tests modifying that and I missed it?